ipn/ipnlocal/local.go

```diff
@@ -1690,24 +1691,70 @@ func (b *LocalBackend) readPoller() {
 	}
 }
 
-// send delivers n to the connected frontend. If no frontend is
-// connected, the notification is dropped without being delivered.
-func (b *LocalBackend) send(n ipn.Notify) {
+// WatchNotifications subscribes to the ipn.Notify message bus notification
+// messages.
+//
+// WatchNotifications blocks until ctx is done.
+//
+// The provided fn will only be called with non-nil pointers. The caller must
+// not modify roNotify. If fn returns false, the watch also stops.
+//
+// Failure to consume a many notifications will result in dropped notifications.
+// There is currently (2022-11-22) no mechanism provided to detect when a
+// message has been dropped.
+func (b *LocalBackend) WatchNotifications(ctx context.Context, fn func(roNotify *ipn.Notify) (keepGoing bool)) {
```

Comment:
make a view?

Reply:
I don't think it works.

Also, this is all temporary. And the read-only rules only apply to within the tailscaled process. Once they come out the other side in user code then it's owned by the caller.